src/main/java/org/wildfly/security/password/impl/MaskedPasswordImpl.java

```diff
@@ -206,7 +207,7 @@ public boolean equals(final Object obj) {
             return false;
         }
         MaskedPasswordImpl other = (MaskedPasswordImpl) obj;
-        return iterationCount == other.iterationCount && Arrays.equals(initialKeyMaterial, other.initialKeyMaterial) && Arrays.equals(salt, other.salt) && Arrays.equals(maskedPasswordBytes, other.maskedPasswordBytes) && algorithm.equals(other.algorithm);
+        return iterationCount == other.iterationCount && Arrays.equals(initialKeyMaterial, other.initialKeyMaterial) && MessageDigest.isEqual(salt, other.salt) && MessageDigest.isEqual(maskedPasswordBytes, other.maskedPasswordBytes) && algorithm.equals(other.algorithm);
```

Comment:
Is this "Arrays.equals" intentional?

Reply:
It is. Based on https://github.com/wildfly-security/wildfly-elytron/pull/1779.